kibicara/platforms/email/webapi.py

```diff
@@ -0,0 +1,151 @@
+
+
+@hood_router.delete('/', status_code=status.HTTP_200_OK)
+async def email_delete(hood=Depends(get_hood)):
+    """ Delete an Email bot. Call this when deleting a hood.
+    Stops and deletes the Email bot as well as all subscribers.
+
+    :param hood: Hood the Email bot belongs to.
+    """
+    email_bot = await Email.objects.get(hood=hood.id)
+    spawner.stop(email_bot)
+    await EmailSubscribers.objects.delete_many(hood=hood.id)
+    await email_bot.delete()
+
+
+@hood_router.post('/subscribe/')
+async def email_subscribe(subscriber: Subscriber, hood=Depends(get_hood)):
+    """ Send a confirmation mail to subscribe to messages via email.
+
+    :param subscriber: Subscriber object, holds the email address.
+    :param hood: Hood the Email bot belongs to.
+    :return: Returns status code 200 after sending confirmation email.
+    """
+    secretbox = SecretBox(Email.secret)
+    token = secretbox.encrypt({'email': subscriber.email, }, encoder=URLSafeBase64Encoder)
+    asciitoken = token.decode('ascii')
+    confirm_link = (
+        config['root_url'] + "api/" + hood.id + "/email/subscribe/confirm/" + asciitoken
+    )
+    send_email(
+        subscriber.email,
+        "Subscribe to Kibicara " + hood.name,
+        sender=hood.name,
+        body="To confirm your subscription, follow this link: " + confirm_link,
+    )
+    return status.HTTP_200_OK
+
+
+@hood_router.post('/subscribe/confirm/{token}')
+async def email_subscribe_confirm(token, hood=Depends(get_hood)):
+    """ Confirm a new subscriber and add them to the database.
+
+    :param token: encrypted JSON token, holds the email of the subscriber.
+    :param hood: Hood the Email bot belongs to.
+    :return: Returns status code 200 after adding the subscriber to the database.
+    """
+    secretbox = SecretBox(Email.secret)
+    json = secretbox.decrypt(token.encode('ascii'), encoder=URLSafeBase64Encoder)
+    try:
+        await EmailSubscribers.objects.create(hood=hood.id, email=json['email'])
+        return status.HTTP_201_CREATED
+    except IntegrityError:
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT)
+
+
+@hood_router.get('/unsubscribe/{token}', status_code=status.HTTP_200_OK)
+async def email_unsubscribe(token, hood=Depends(get_hood)):
+    """ Remove a subscriber from the database when they click on an unsubscribe link.
+
+    :param token: encrypted JSON token, holds subscriber email + hood.id.
+    :param hood: Hood the Email bot belongs to.
+    """
+    secretbox = SecretBox(Email.secret)
+    json = secretbox.decrypt(token.encode('ascii'), encoder=URLSafeBase64Encoder)
+    # If token.hood and url.hood are different, raise an error:
+    if hood.id is not json['hood']:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST)
+    await EmailSubscribers.objects.delete_many(hood=json['hood'], email=json['email'])
+
+
+@mailbox_router.post('/messages/')
+async def email_message_create(message: BodyMessage):
```

Comment:
This could become part of hood_router if get_email_row() was moved to kibicara_mda.py.

Reply:
Then we could remove the mailbox_router completely.